frontend/src/components/ProtectedRoutes.tsx

```diff
@@ -0,0 +1,13 @@
+import { Navigate, Outlet } from 'react-router-dom';
+import { useAuthContext } from '../context/AuthContext';
+
+
+export default function ProtectedRoutes() {
+    const { user, loading } = useAuthContext();
+
+    if (loading) {
+        return <div>Loading...</div>; // Or a more sophisticated loading spinner
```

Comment:
```suggestion
        return <div>Loading...</div>; // Or a more sophisticated loading spinner
```
remove this comment or mark it as TODO